cpp/src/structs/utilities.cpp

```diff
@@ -309,6 +309,11 @@ void superimpose_parent_nulls(bitmask_type const* parent_null_mask,
                               rmm::cuda_stream_view stream,
                               rmm::mr::device_memory_resource* mr)
 {
+  if (child.type().id() == cudf::type_id::EMPTY) {
```

Comment:
Although without actual effect, I would recommend this is a better pattern to check data type in libcudf:
```suggestion
  if (child.type() == data_type{type_id::EMPTY}) {
```

Reply:
There's other places within this file that check for `type().id() == type_id::STRUCT`. Is `STRUCT` a special case?

I don't mind this change at all, it just feels a bit weird to me to change it just in this one place in this file.

Reply:
I see. My take was we should encourage use of defined operator `==` for `data_type`. Although for now the only difference it has are with `fixed_point` types. I don't strongly insist though considering it involves an extra comparison.